src/easylink/step.py

```diff
@@ -1,34 +1,139 @@
-from dataclasses import dataclass
-from typing import TYPE_CHECKING, Callable
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import TYPE_CHECKING
 
 from networkx import MultiDiGraph
 
 from easylink.implementation import Implementation
-from easylink.utilities.validation_utils import validate_input_file_dummy
 
 if TYPE_CHECKING:
     from easylink.configuration import Config
 
 
-@dataclass
-class Step:
+class Step(ABC):
+    @abstractmethod
+    def get_subgraph(self, config: "Config") -> MultiDiGraph:
+        pass
+
+
+class CompositeStep(Step):
+    def __init__(self, name: str, **params) -> None:
```

Comment:
again, not super happy to be using unguarded **kwargs type instantiation for these different steps. Wondering if there is a better way